genkit-tools/common/src/types/apis.ts

```diff
@@ -150,3 +150,25 @@ export const RunNewEvaluationRequestSchema = z.object({
 export type RunNewEvaluationRequest = z.infer<
   typeof RunNewEvaluationRequestSchema
 >;
+
+export const ValidateDataRequestSchema = z.object({
+  dataSource: z.object({
+    datasetId: z.string().optional(),
+    data: InferenceDatasetSchema.optional(),
+  }),
+  actionRef: z.string(),
+});
+export type ValidateDataRequest = z.infer<typeof ValidateDataRequestSchema>;
+
+export const ValidateDataResponseSchema = z.object({
+  valid: z.boolean(),
+  datasetErrors: z
+    .record(z.string(), z.any())
+    .describe('Errors mapping when validating dataset')
+    .optional(),
+  dataErrors: z
+    .record(z.string(), z.any())
+    .describe('Errors mapping when validating raw data')
+    .optional(),
```

Comment:
is it necessary to separate these? I mean, at the end of the day it's a map of test case identifier (or a row index, which is the same thing) to errors. Overloading inputs is OK, but overloading output should be minimized.